### ISSUE TITLE
support for TypeScript 2.0

### ISSUE THREAD

**Issue by ShawnTalbert** [headintheclouddev/typings-suitescript-2.0#4]
Have we considered how this would be used in TypeScript 2.0 and its new handling of [typings](https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/)?

Basically, when TS 2.0.x and NS 2016.2 are released I hope to finalize [NFT-SS2](https://github.com/ExploreConsulting/netsuite-fasttrack-toolkit-ss2) for initial public release and would love to consolidate on a single typings implementation for these NetSuite modules. Ideally the whole enchilada would be installable with npm alone.

**Comment by johnogle222** [comment]
Agreed I would love to do this just through npm. From what I've gathered the way to move forward would be to get this published on the DefinitelyTyped repository. What I'm not sure is if we need to capture the "N" folder due to NetSuite's namespace or if we would be able to add to the existing suitescript folder from @darrenhillconsulting. Does anyone more familiar with TS2.0 module resolution know the answer to this?

**Comment by ShawnTalbert** [comment]
There are a couple new configuration options in TS2.0 that should smooth this over nicely. SuiteScript2.0 also supports require.config() but I'd like to avoid introducing that extra complexity unless we really need it.  My gut feel (before analyzing it in depth) is that we can leverage:

*  TS2.0 config settings to handle compile-time resolution for all code and dependencies 
*  NetSuite SS2.0 require.config settings to support multiple versions of libraries (i.e. resolve run-time path issues).

**Comment by ShawnTalbert** [comment]
I've done some experimentation with TS 2.0.0 and module resolution on this topic.

A workable sample can be achieved using the following steps in a fresh directory:

1. `npm install github:headintheclouddev/typings-suitescript-2.0`
2. remove the `/// <reference path="typings/index.d.ts" />` from **N/record.d.ts** and **N/search.d.ts** (we'd want to remove these from all `N/` declaration files because we don't need the _typings_ tool?)
2. create a `tsconfig.json` that looks something like this

 ```json
{
  "compilerOptions": {
    "module": "amd",
    "target": "es5",
    "sourceMap": false,
    "baseUrl": ".",
    "lib":["es5","es2015.promise"],
    "paths": {
      "N/*": [
        "node_modules/@hitc/netsuite-types/N/*"
      ]
    }
  }
}
```

Then the following test TS file will compile without error:

```typescript
import * as record from "N/record"
var r = record.load({type:'customer', id:1 });
```

This is an exciting success, no?  I think the main change to this repo will be dropping the `///ref` tags from all declaration files (resolving #3) , and arguably dropping the dependency on **typings** altogether, or do we need it for something else? 

With those we should be able to leverage this repo for all the N/ typings in NFT-SS2. 

How would you like to move forward? 

p.s. this works fine; perhaps _later_ we could figure out what's needed/feasibility to get this working within the new @types npm namespace?

**Comment by ShawnTalbert** [comment]
Note, the new `lib` option allows us to pull in the Promise definition without installing something separate from typings, provided the TS provided declaration is compatible with NS (haven't tested that).

Hence we eliminate the extra step of installing typings for promises.

**Comment by ShawnTalbert** [comment]
TS 2.0(.3)  is official....

**Comment by johnogle222** [comment]
Haha yes I've been dragging my heels on this. I'll hopefully get this all integrated by the end of the week :)

**Comment by ShawnTalbert** [comment]
I'd just rather run off the official repo rather than our custom fork of it if possible :)

**Comment by johnogle222** [comment]
Closed with #4

Thank you so much, Shawn!
